python/tests/test_mutable_file_echo_api.py

```diff
@@ -129,5 +137,104 @@ def test_subsequent_connection(self):
         self.assertNotEqual(other_uid, other_prev_uid)
 
         # [c] check the contents of the loaded file again in the original connection
-        (prev_uid, state) = self.assertShow(c, state=state,expected='Hello World!\n')
+        (prev_uid, state) = assertShow(self, c, state=state,expected='Hello World!\n')
+
+
+class InterruptTests(unittest.TestCase):
+    # Connection to server
+    c = None
+    # process running the server
+    p = None
+
+    @classmethod
+    def setUpClass(self):
+        p = subprocess.Popen(
+            ["cabal", "run", "exe:mutable-file-echo-api", "--verbose=0", "--", "--max-occupancy", "2", "http", "/", "--port", "8083", "--file", str(hello_file)],
+            stdout=subprocess.PIPE,
+            stdin=subprocess.DEVNULL,
+            stderr=subprocess.PIPE,
+            start_new_session=True)
+        time.sleep(3)
+        assert(p is not None)
+        poll_result = p.poll()
+        if poll_result is not None:
+            print(poll_result)
+            print(p.stdout.read())
+            print(p.stderr.read())
+        assert(poll_result is None)
+
+        self.p = p
+
+    @classmethod
+    def tearDownClass(self):
+        os.killpg(os.getpgid(self.p.pid), signal.SIGKILL)
+        super().tearDownClass()
 
+    # to be implemented by classes extending this one
+    def test_interrupts(self):
+        c1 = argo.ServerConnection(argo.HttpProcess('http://localhost:8083/'))
+        c2 = argo.ServerConnection(argo.HttpProcess('http://localhost:8083/'))
```

Comment:
Similarly, can the `HttpProcess` be shared here?

Reply:
Negative for same reason noted above.